### ISSUE TITLE
bug: fail build vue app in last update with vite

### ISSUE THREAD

**Issue by KANekT** [ionic-team/ionic-framework#25485]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [x] Nightly

### Current Behavior

Error after update to last version in vue, example file:

`node_modules/@ionic/vue/dist/types/components/IonIcon.d.ts`

in 6.1.10-nightly.20220614
```
export declare const IonIcon: import("vue").DefineComponent<{
    color: StringConstructor;
    flipRtl: BooleanConstructor;
    icon: StringConstructor;
    ios: StringConstructor;
    lazy: StringConstructor;
    md: StringConstructor;
    mode: StringConstructor;
    name: StringConstructor;
    size: StringConstructor;
    src: StringConstructor;
}, () => import("vue").VNode<import("vue").RendererNode, import("vue").RendererElement, {
    [key: string]: any;
}>, unknown, {}, {}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, Record<string, any>, string, import("vue").VNodeProps & import("vue").AllowedComponentProps & import("vue").ComponentCustomProps, Readonly<{} & {
    lazy?: string;
    mode?: string;
    ios?: string;
    md?: string;
    name?: string;
    color?: string;
    size?: string;
    icon?: string;
    src?: string;
    flipRtl?: boolean;
}>, {}>;
```

in 6.1.10-nightly.20220615

```
export declare const IonIcon: import("vue").DefineComponent<{
    color: StringConstructor;
    flipRtl: BooleanConstructor;
    icon: StringConstructor;
    ios: StringConstructor;
    lazy: StringConstructor;
    md: StringConstructor;
    mode: StringConstructor;
    name: StringConstructor;
    size: StringConstructor;
    src: StringConstructor;
}, () => import("vue").VNode<import("vue").RendererNode, import("vue").RendererElement, {
    [key: string]: any;
}>, unknown, {}, {}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, Record<string, any>, string, import("vue").VNodeProps & import("vue").AllowedComponentProps & import("vue").ComponentCustomProps, Readonly<import("vue").ExtractPropTypes<{
    color: StringConstructor;
    flipRtl: BooleanConstructor;
    icon: StringConstructor;
    ios: StringConstructor;
    lazy: StringConstructor;
    md: StringConstructor;
    mode: StringConstructor;
    name: StringConstructor;
    size: StringConstructor;
    src: StringConstructor;
}>>, {
    flipRtl: boolean;
}>;
```

### Expected Behavior

`<ion-icon slot="icon-only" :icon="icons.close" />`

Type string is not assignable to type InferPropType<StringConstructor>
Type string is not assignable to type [StringConstructor] extends [Prop<infer V, infer D>] ? (unknown extends infer V ? IfAny<V, V, D> : V) : StringConstructor     
Type string is not assignable to type StringConstructor

### Steps to Reproduce

vue-tsc --noEmit && vite build

### Ionic Info

Ionic:

   Ionic CLI       : 6.18.1 (C:\Users\KANekT\AppData\Roaming\npm\node_modules\@ionic\cli)

Capacitor:

   Capacitor CLI      : 3.5.1
   @capacitor/android : 3.5.1
   @capacitor/core    : 3.5.1
   @capacitor/ios     : 3.5.1

Utility:

   cordova-res : not installed globally
   native-run  : 1.6.0

System:

   NodeJS : v16.15.1 (C:\Program Files\nodejs\node.exe)
   npm    : 8.11.0
   OS     : Windows 10

**Comment by KANekT** [comment]
Vue 3

**Comment by reslear** [comment]
@KANekT это очевидно что 3, полная версия, и версия vite?

**Comment by reslear** [comment]
hi @liamdebeasi I have a similar problem, but not with build, but with volar vscode ext
<img width="1423" alt="Screenshot 2022-06-16 at 17 01 23" src="https://user-images.githubusercontent.com/12596485/174087197-be6f1951-ec34-4704-b43f-f25414a584a1.png">
repro https://github.com/reslear/ionic-vue-pnpm-express-monorepo

**Comment by liamdebeasi** [comment]
Ionic Vue does not support Volar. See: https://github.com/ionic-team/ionic-framework/issues/24169

**Comment by KANekT** [comment]
https://github.com/KANekT/ion-menu-bug

i have warning in webstorm
![изображение](https://user-images.githubusercontent.com/580273/174269711-864a7d01-f958-4c8e-bf45-8a191965fd50.png)

**Comment by liamdebeasi** [comment]
Can everyone try the following dev build and let me know if it resolve the issue?

```
npm install @ionic/vue@6.1.11-dev.11655478768.115de3ec @ionic/vue-router@6.1.11-dev.11655478768.115de3ec
```

**Comment by KANekT** [comment]
Yes it resolve this error

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/25499, and a fix will be available in an upcoming release of Ionic Framework.

**Comment by kachidk** [comment]
> Ionic Vue does not support Volar. See: https://github.com/ionic-team/ionic-framework/issues/24169

Is there a recommendation on what to use instead?

**Comment by reslear** [comment]
@kachidk volar is support, but without html types, i'm use everyday - no problems

**Comment by kachidk** [comment]
> I have a similar problem, but not with build, but with volar vscode ext
<img width="1423" alt="174087197-be6f1951-ec34-4704-b43f-f25414a584a1" src="https://user-images.githubusercontent.com/61981167/175023607-cf70cb11-2cbf-455e-967c-44d11b606723.png">

I have this same problem similar to you have did you solve it

**Comment by liamdebeasi** [comment]
Hi everyone,

This issue was resolved in Ionic 6.1.11. Please update to the latest version of Ionic Vue to receive a fix for this issue.

**Comment by liamdebeasi** [comment]
Also, as per my comment in https://github.com/ionic-team/ionic-framework/issues/25485#issuecomment-1157699538, Volar support should be improved in Ionic 6.1.11. Please give it a try and let us know if you run into any issues on https://github.com/ionic-team/ionic-framework/issues/24169.
